nilearn/regions/rena_clustering.py

```diff
@@ -0,0 +1,580 @@
+"""Recursive Neighbor Agglomeration (ReNA):
+    fastclustering for approximation of structured signals
+"""
+# Author: Andres Hoyos idrobo, Gael Varoquaux, Jonas Kahn and  Bertrand Thirion
+# License: simplified BSD
+
+import numpy as np
+import warnings
+from scipy.sparse import csgraph, coo_matrix, dia_matrix
+from sklearn.externals.joblib import Memory
```

Comment:
```suggestion
try:
    from sklearn.externals.joblib import Memory
except ImportError:
    from joblib import Memory
```

Joblib is being unvendored from scikit-learn. We will need to support both imports path for a little while.